src/main/java/org/opentripplanner/routing/algorithm/transferoptimization/services/TransitPathLegSelector.java

```diff
@@ -9,18 +9,29 @@
 
 
 /**
- * This class takes a list of transit legs and returns the best leg based on the
- * {@link TransferOptimizedFilterFactory} and the min-time-limit. The leg arrival-time must be
- * AFTER the min-time-limit. The filter is used to pick the best leg from the legs satisfying the
- * min-time-limit.
+ * This class takes a list of transit legs and returns the best leg based on the {@link
+ * TransferOptimizedFilterFactory} and the earliest-boarding-time. The filter is used to pick the
+ * best leg from the legs witch can be boarded after the earliest-boarding-time.
+ * <p>
+ * HOW IT WORKS
+ * <p>
+ * The initial set of elements are put in the "reminding" set, and the "selected" set is empty.
+ * <p>
+ * Each time the {@link #next(int)} method is called the reminding elements are search for
+ * candidates where boarding is possible. The new candidates are added to the selected set and the
+ * selected set is filtered, and are elements dropped. Dropped elements represent none optimal
```

Comment:
Should this read like this?
```suggestion
 * selected set is filtered, and remaining elements are dropped. Dropped elements represent no optimal
```